File: packages/rust-api/src/enums/exercise_type.rs
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, Deserialize, Serialize, sqlx::Type)]
#[serde(rename_all = "snake_case")]
#[sqlx(type_name = "varchar")]
#[sqlx(rename_all = "snake_case")]
pub enum ExerciseType {
    Activation,
    Class,
    Endurance,
    Strength,
}

File: packages/app/src/enums/ExerciseType.ts
const ExerciseType = {
    Activation: 'activation',
    Class: 'class',
    Endurance: 'endurance',
    Strength: 'strength',
    Stretch: 'stretch',
} as const;

type ExerciseType = typeof ExerciseType[keyof typeof ExerciseType];

export default ExerciseType;
